src/main/java/net/cardosi/mojo/cache/CachedProject.java

```diff
@@ -953,7 +941,13 @@ public FileVisitResult preVisitDirectory(Path path, BasicFileAttributes basicFil
 
             @Override
             public FileVisitResult visitFile(Path path, BasicFileAttributes basicFileAttributes) throws IOException {
-                hash.append(Files.readAllBytes(path));
+                byte[] bytes = diskCache.getHashes().get(path);
+                if ( bytes != null) {
+                    hash.append(bytes);
+                } else {
+                    hash.append(Files.readAllBytes(path));
+                }
```

Comment:
well this isn't ideal - if the getHashes() map is populated with an item, we get one value, otherwise we get another - shouldn't we wrap the Files.getAllBytes call in a call to take the murmur hash of the file?

Reply:
Perhaps `FileHasher.DEFAULT_FILE_HASHER.hash(path).toString()`? to replace the Files.readAllBytes call? Still getting the hang of this

Reply:
I actually think it may be better to throw an exception of the cache is null, as I'm not sure that should ever happen.

Reply:
ok change made. Let's see if the exception is ever thrown and we can look into why.